packages/node/src/executors/execute/execute.impl.spec.ts

```diff
@@ -329,4 +342,230 @@ describe('NodeExecuteBuilder', () => {
+          done: false,
+          value: {
+            outfile: 'outfile.js',
+            success: true,
+          },
+        });
+
+        const result = await Promise.race([
+          executorIterator.next(),
+          new Promise(() => {
+            mockSubProcessExit(1);
+            events.push(firstEvent);
+          }),
+        ]);
+
+        events.push(secondEvent);
+
+        expect(result).toEqual({
+          value: undefined,
+          done: true,
+        });
+        expect(events).toEqual([firstEvent, secondEvent]);
+      });
+
+      it('When sub process emits, it should propagate when configured', async () => {
+        const expectedExitCode = 42;
+        const executorIterator = executeExecutor(
+          {
+            ...testOptions,
+            watch: false,
+            emitSubprocessEvents: true,
+          },
+          context
+        );
+
+        expect(await executorIterator.next()).toEqual({
+          done: false,
+          value: {
+            outfile: 'outfile.js',
+            success: true,
+          },
+        });
+
+        mockSubProcessExit(expectedExitCode);
+
+        expect(await executorIterator.next()).toEqual({
+          done: false,
+          value: {
+            exitCode: expectedExitCode,
+          },
+        });
+      });
+
+      it('Should propagate sub process events in expected order', async () => {
+        const expectedSubProcessEvents = [
+          {
+            message: 'hello',
+          },
+          {
+            message: 'world',
+          },
+          {
+            error: new Error('error'),
+          },
+          {
+            exitCode: 1,
+          },
+        ];
+        const expectedTotalEvents = 5;
+
+        let subProcessEvents = [];
+        let totalEvents = 0;
+
+        const doSubProcessWork = () => {
+          mockSubProcessMessage('hello');
+          mockSubProcessMessage('world');
+          mockSubProcessError(new Error('error'));
+          mockSubProcessExit(1);
+        };
+
+        for await (const event of executeExecutor(
+          {
+            ...testOptions,
+            watch: false,
+            emitSubprocessEvents: true,
+          },
+          context
+        )) {
+          if (totalEvents === 0) {
+            expect(event).toEqual({
+              outfile: 'outfile.js',
+              success: true,
+            });
+            doSubProcessWork();
+          } else {
+            subProcessEvents.push(event);
+          }
+
+          totalEvents += 1;
+        }
+
+        expect(subProcessEvents).toEqual(expectedSubProcessEvents);
+        expect(totalEvents).toEqual(expectedTotalEvents);
+      });
+    });
+
+    describe('true', () => {
+      beforeEach(() => {
+        (devkit.runExecutor as any).mockImplementation(async function* () {
+          yield { success: true, outfile: 'outfile.js' };
+          await new Promise((resolve) => {
+            setTimeout(resolve, 100);
```

Comment:
```suggestion
            setTimeout(resolve, 1);
```
Do you need to wait a whole 100ms here?